docs/engineering-defaults.md

```diff
@@ -1,7 +1,147 @@
+engineering teams.
+
 ## Pair Programming
 
-## Trunk Based Development
+Pair programming is a software development technique in which two developers work
+together at one workstation. One, the driver, writes code while the other, the
+observer or navigator, reviews each line of code as it is typed in. The two
+developers switch roles frequently.
+
+This method provides extremely fast feedback to developers about their code
+changes and design decisions before. In turn, this prevents wasted effort and produces
+cleaner, simpler code at a faster rate.
+
+Tooling can greatly improve the developers' experience while pair programming.
+Audio calls and screen sharing enable pair programming for engineers in
+different physical locations. Similarly, IDE tools such as Visual Studio Code's
+[Live Share][liveshare] extension allow both parties to develop code changes
+simultaneously in a single workspace.
 
 ## Small Batch Delivery
+
+Open O11y encourages teams to deliver in small batches. Planning small portions
+of work enables teams to deploy, demo, and request feedback more frequently.
+Much like pair programming, this practice aids teams in coordinating their
+efforts and safeguards against individuals investing time and energy in
+developing an incorrect solution.
+
+## Trunk Based Development
+
+> A source-control branching model, where developers collaborate on code in a
+> single branch called 'trunk'[^trunksummary]
+
+Open O11y highly recommends using [Trunk Based Development][trunk] over other
+branching models. Its focus on simplicity offers many advantages:
+
+* The 'main' branch represents the application and is always production-ready
+* Smaller, short-lived branches are dramatically safer to merge
+* Developers enjoy a reduced cognitive load
+* CI/CD requires less maintenance and/or is much simpler
+
+[Gitflow][gitflow]
+
+## CI/CD
+
+### Continuous Integration
+
+> Continuous Integration (CI) is a development practice where developers
+> integrate code into a shared repository frequently, preferably several times a
+> day. Each integration can then be verified by an automated build and automated
+> tests.[^ci]
+
+CI is a critical part of the development lifecycle and is a natural compliment
+to [Small Batch Delivery](#small-batch-delivery). This provides developers with
+very frequent feedback on their code changes. When problematic merges occur,
+teams are quickly notified. More importantly, a bad merge can be identified
+immediately, allowing the relevant team members to quickly fix the issue.
+
+### Continuous Delivery
+
+Continuous Delivery can be traced back to the first principal of the Agile
+Manifesto[^agilemanifesto] and naturally follows Continuous Integration. It is
+the practice of sequentially deploying every viable version of an application,
+ideally to production. This ensures that the software can always be released at
+any time.
+
+### Speed
```

Comment:
Was this Focus on Flow before?

Reply:
This came from the "Fast Automated Builds" portion of the Engineering Defaults google doc. There's so much overlap with ci/cd. So I moved the basic idea here.

I think we can create a better title. Automation Performance or similar.

Reply:
Why not keep it as "Fast Automated Builds?"

Reply:
I thought the "automated" aspect was redundant with the previous sections. No good reason for changing "fast" to "speed" though.